data/integration_tests/percolator/IntegrationTest_Percolator_Speed.py

```diff
@@ -0,0 +1,242 @@
+        buffer = site.read(block_sz)
+        if not buffer:
+            break
+        file_size_dl += len(buffer)
+        f.write(buffer)
+        downloadProgress = file_size_dl * 100. / file_size
+        status = r"%10d  [%.0f%%]" % (file_size_dl, downloadProgress)
+        status = status + chr(8)*(len(status)+1)
+        if (downloadProgress - prevDownloadProgress > 1. or downloadProgress > 99.9):
+            print (status, end="")
+            prevDownloadProgress = downloadProgress
+    f.close()
+    print("\nDownload complete.\n")
+
+def extractTarFile(fname):
+    if fname.endswith("tar.gz"):
+        tar = tarfile.open(fname, "r:gz")
+        tar.extractall(testDataFolderPath)
+        tar.close()
+    elif fname.endswith("tar"):
+        tar = tarfile.open(fname, "r:")
+        tar.extractall(testDataFolderPath)
+        tar.close()
+
+def getArguments():
+    parser = ArgumentParser(description="Measure speed of perculator application.")
+    parser._action_groups.pop()
+    required = parser.add_argument_group('required arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    required.add_argument('-d','--data', type=str, default = "" , metavar='', required=False, help="Path to input-data used by percolator.")
+    required.add_argument('-r','--runs', type=int, metavar='', default=3, required=False, help="Number of tests to evaluate.")
+    optional.add_argument('-f','--flags', type=str, default = "" , metavar='', required=False, help="Flags to use with the executable.")
+    optional.add_argument('-a','--await_tests', default=False, action="store_true", required=False, help="Wait for others processes running this script to finish before starting.")
+    optional.add_argument('-p','--processes', nargs='+', type=int, metavar='',required=False, help="Other test-PIDs to wait for before starting test(s).")
+    optional.add_argument('-c','--comments', type=str, default="", metavar='', required=False, help="Comments regarding test details.")
+    return parser.parse_args()
+
+def awaitOtherTasks(args):
+    if(args.processes is not None):
+        waitForProcesses(args.processes)
+    if(args.await_tests is True):
+        waitForOtherTests()
+
+def waitForProcess(pid):
+    numIterations = 0
+    delayMultiplier = 1
+    while(True):
+        if (psutil.pid_exists(pid)):
+            if(numIterations == 0 or numIterations > 60*delayMultiplier):
+                delayMultiplier = delayMultiplier * 2
+                numIterations = 1
+                print("Waiting for PID " + str(pid) + " to finish. (" + datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S") + ")")
+            time.sleep(45)
+        else:
+            return
+        numIterations += 1
+
+def waitForProcesses(pids):
+    if (pids is None):
+        return
+    for p in pids:
+        waitForProcess(p)
+
+def waitForOtherTests(): #This could cause the computer to wait indefinately if stuck in a race condition. Use with caution.
```

Comment:
I feel like this comment belongs on `waitForProcess`, since it can also loop forever. Is there some reason the tests are especially susceptible to race conditions? (Also, spelling of "indefinitely".)